core/components/molecules/keyValuePair/ValueElement.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { IconProps } from '@/index.type';
+import { Icon, Text } from '@/index';
+import classNames from 'classnames';
+import { BaseProps, extractBaseProps } from '@/utils/types';
+
+type IconPosition = 'left' | 'right';
+
+export interface ValueElementProps extends BaseProps {
+  children: React.ReactNode;
+  icon?: string;
+  iconOptions?: IconProps;
+  iconPosition?: IconPosition;
```

Comment:
value should support only text in value prop and everything else should be passed as children, keep only two props value and children and both optional.